### PR TITLE
Remember section expanded state.

### LINKED ISSUE #149: Sections should remember last state (collapsed or expanded)
When one open the Manage Connection hub in Team Explorer, the github node comes up at top, which is fine, but it always comes up expanded, which can be sort of annoying if one wants quickly to see what is further down.
The github node should either:  1) start collapsed  or 2) remember if one sets it collapsed or expanded for the next time the same hub is opened.

### PR DESCRIPTION
**Pull request by grokys**
Fixes #149 .

Sections in the Connect pane in Team Explorer now remember their expanded state between Visual Studio sessions.

### CODE REVIEW

**Comment by shana** [comment]
@grokys Whenever this is ready for review, make a comment to that effect (with copious use of emoji like 🚀  or 🍍  or 👀 )

**Comment by grokys** [comment]
I think this is ready for :ear_of_rice: :earth_africa: :facepunch: review!

File: src/GitHub.Exports/Settings/UIState.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    /// <summary>
+    /// Stores persistent UI state in <see cref="IPackageSettings"/>.
+    /// </summary>
+    public class UIState
+    {
+        /// <summary>
+        /// Gets or sets the UI state for the <see cref="IGitHubConnectSection"/>s in Team Explorer.
+        /// </summary>
+        public List<GitHubConnectSectionState> GitHubConnectSections { get; set; }
+            = new List<GitHubConnectSectionState>();
```
Comment (shana) [inline src/GitHub.Exports/Settings/UIState.cs]: Love this syntax and keep forgetting this is possible!

File: src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs [hunk 2]
```diff
@@ -28,6 +29,9 @@ public class GitHubConnectSection : TeamExplorerSectionBase, IGitHubConnectSecti
         readonly int sectionIndex;
         bool isCloning;
         bool isCreating;
+        IPackageSettings packageSettings;
```
Comment (shana) [inline src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs]: This one can probably be `readonly`

**Comment by shana** [comment]
This is great, awesome job! :tada:

**Pull request merged**